### PR TITLE
Raw control of the drone

### PR DESCRIPTION
**Pull request by bjornborg**
This pull requests combines basic dronstate + basic pid controller + ps3 controller proxy.

### CODE REVIEW

File: cmake.Modules/CompileFlags.cmake [hunk 1]
```diff
@@ -21,7 +21,7 @@ if("${CMAKE_SYSTEM_NAME}" STREQUAL "Darwin")
     set(C_COMPILER_FLAGS "-std=c99")
     set(CXX_COMPILER_FLAGS "-std=c++14")
 elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux")
-    set(COMPILER_FLAGS "-fstack-protector-all -fPIC -ggdb -Wall -Weffc++ -Werror -Wextra -Wshadow -pedantic-errors")
+    set(COMPILER_FLAGS "-fstack-protector-all -fPIC -ggdb -Wall -Weffc++ -Werror -Wextra -Wshadow -pedantic-errors -pthread")
```
Comment (bjornborg) [inline cmake.Modules/CompileFlags.cmake]: @proundeadzx Is pthread still being used?

File: code/proxy-miniature/ps3controller/src/PS3Controller.cpp [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+
+namespace opendlv {
+namespace proxy {
+namespace miniature {
+
+int latestJsNumber = 0;
+int latestJsValue = 0;
+double baseThrust = -1.0;
+opendlv::logic::StateRequest stateReq(-1.0,0.0,0.0, 0.0);
+
```
Comment (bjornborg) [inline code/proxy-miniature/ps3controller/src/PS3Controller.cpp]: @proundeadzx Shouldn't these brand under class variables? Right now they are exposed...

File: code/proxy-miniature/ps3controller/src/PS3Controller.cpp [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+  close(m_ps3controllerDevice);
+  latestJsNumber = 3;
+  latestJsValue = 1;
+  sendStateRequest();
+  latestJsNumber = 2;
+  latestJsNumber = 0;
+  sendStateRequest();
+  latestJsNumber = 3;
+  latestJsValue = 0;
+  sendStateRequest();
```
Comment (bjornborg) [inline code/proxy-miniature/ps3controller/src/PS3Controller.cpp]: @proundeadzx What happens here?

File: code/proxy-miniature/ps3controller/src/PS3Controller.cpp [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+  }else if (latestJsNumber == 3 && latestJsValue > 0) {
+    stateReq.setPitch(-((double)latestJsValue/32767)*(M_PI/36));
+  }else {
+    stateReq.setPitch(0);
+  }
+
+  odcore::data::Container c(stateReq);
+  getConference().send(c);
+  if (m_debug) {
+   cout << stateReq.toString() << endl;
```
Comment (bjornborg) [inline code/proxy-miniature/ps3controller/src/PS3Controller.cpp]: Make sure that indentation is 2 spaces :)

File: cmake.Modules/CompileFlags.cmake (hunk 1 above)
Comment (proundeadzx) [reply cmake.Modules/CompileFlags.cmake]: No, my implementation don't use multithreading so the flag is not needed

File: code/proxy-miniature/ps3controller/src/PS3Controller.cpp (hunk 2 above)
Comment (proundeadzx) [reply code/proxy-miniature/ps3controller/src/PS3Controller.cpp]: That is true, I forgot to do that

File: code/proxy-miniature/ps3controller/src/PS3Controller.cpp (hunk 3 above)
Comment (proundeadzx) [reply code/proxy-miniature/ps3controller/src/PS3Controller.cpp]: This sets all the roll, pitch, and base throttle to zero at teardown.

File: code/proxy-miniature/ps3controller/src/PS3Controller.cpp (hunk 4 above)
Comment (proundeadzx) [reply code/proxy-miniature/ps3controller/src/PS3Controller.cpp]: whoops

**Pull request merged**